eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java

```diff
@@ -818,6 +818,13 @@ else if (replaceUnderscore && !t.getName().matches(EwonConsts.ALLOWED_TAGNAME_RE
                             // Register tag write handle with provider
                             provider.registerWriteHandler(p, new WriteHandler() {
                                 public QualityCode write(TagPath tagPath, Object o) {
+                                    Object currentTagValue;
+                                    try {
+                                        currentTagValue = gatewayContext.getTagManager().readAsync(
+                                            Collections.singletonList(tagPath)).get().get(0).getValue();
+                                    } catch (Exception e) {
+                                        currentTagValue = null;
```

Comment:
What are the implications of setting the currentTagValue to null?  is it fine to do this?

Reply:
If it's null, the value just gets set to the default of that type, so int: 0, boolean: false, etc.